apps/web-giddh/src/app/contact/payment-aside/payment-aside.component.ts

```diff
@@ -161,4 +271,192 @@ export class PaymentAsideComponent implements OnInit {
+            })
+        }
+    }
+
+    /**
+        *To select bank event
+        *
+        * @param {*} event selected bank object
+        * @memberof PaymentAsideComponent
+        */
+    public bulkPayVendor(): void {
+        this._companyService.bulkVendorPayment(this.companyUniqueName, this.requestObjectTogetOTP).subscribe(response => {
+
+            if (response.status === 'success') {
+                this._toaster.successToast(response.body);
+            } else if (response.status === 'error') {
+                this._toaster.errorToast(response.message, response.code);
+            }
+        });
+    }
+
+
+
+
+    public getTotalAmount(selectedAccount: any[]): void {
+        this.totalSelectedAccountAmount = 0;
+        if (selectedAccount && selectedAccount.length) {
+            this.totalSelectedAccountAmount = selectedAccount.reduce((prev, cur) => {
+                return prev + cur.closingBalanceAmount;
+            }, 0);
+        }
+    }
+
+    /**
+     * To Prepare payment request object and API call
+     *
+     * @param {NgForm} form Form data
+     * @memberof PaymentAsideComponent
+     */
+    public payClicked(form: NgForm) {
+        this.isPayclicked = true;
+        this.prepareRequestObject();
+        this.bulkPayVendor();
+    }
+
+    public prepareRequestObject() {
+        this.requestObjectTogetOTP.bankPaymentTransactions = [];
+        this.requestObjectTogetOTP.totalAmount = String(this.totalSelectedAccountAmount);
+        this.selectedAccForBulkPayment.forEach(item => {
+            let transaction: BankTransactionForOTP = {
+                amount: '',
+                remarks: '',
+                vendorUniqueName: ''
+            };
+            transaction.amount = item.closingBalance.amount;
+            transaction.remarks = item.remarks;
+            transaction.vendorUniqueName = item.uniqueName;
+            this.requestObjectTogetOTP.bankPaymentTransactions.push(transaction);
+        });
+    }
+
+    /**
+     * To cancel OTP request and count down timer
+     *
+     * @memberof PaymentAsideComponent
+     */
+    public clickedCancelOtp() {
+        this.isPayclicked = false;
+        this.timerOn = false;
+        clearTimeout(this.countDownTimerRef);
+    }
+
+    /**
+     * Timer for count down OTP validation API call
+     *
+     * @param {number} remaining   count down remaining time
+     * @returns
+     * @memberof PaymentAsideComponent
+     */
+    public startTimer(remaining: number): any {
+        let min: any = Math.floor(remaining / 60);
+        let sec: any = remaining % 60;
+        min = Number(min) < 10 ? '0' + min : Number(min);
+        sec = Number(sec) < 10 ? '0' + sec : Number(sec);
+        // document.getElementById('timerElement').innerHTML = min + ':' + sec;
+        this.timerCountDown$ = of(min + ':' + sec);
+        remaining -= 1;
+
+        if (remaining >= 0 && this.timerOn) {
+            this.countDownTimerRef = setTimeout(() => {
```

Comment:
Can we not use normal `setInterval` & `clearInterval`?

Reply:
setTimeout we have clear all timer at the add of process